### DISCUSSION TITLE
JanusGraph 1.0.0 release

### DISCUSSION THREAD

**Discussion by ntrhieu89** [JanusGraph/janusgraph#3892] [General]
Hi, 

We are testing the current master branch of JanusGraph and see performance improvements to some of our queries (due to the multiQuery optimizations being used in steps such as hasLabel(), valueMap(), properties, etc..). Hence, we plan to apply it in our production deployment.

Do you have an estimated time for the release of JanusGraph 1.0.0, or any critical features/improvements that are targeted for this release but are missing that we should be aware of?

Thanks.

**Comment by porunov** [comment]
You can always use the latest commit release as JanusGraph produces a release per rach commit (see doc: https://docs.janusgraph.org/advanced-topics/commit-releases/ ).
As for the targeted issues for the release you can find them here: https://github.com/JanusGraph/janusgraph/milestone/21
I proposed the estimated plan for 1.0.0 release in Discord here: https://discord.com/channels/981533699378135051/984361439546441749/1120625697380126720

I think 3.7.0 TinkerPop should be released closer to start of August. After that i t may take a little bit to align everything as proposed in that message. There is no specific or estimated date set to complete the outstanding work because it depends on the contributors activity. 

Below is just copy past of the message from Discord:

I think it makes sense to start preparing JanusGraph for Java 17 builds, sp that on TinkerPop 3.7.0 upgrade we could support Java 17 as well. JanusGraph currently can work with Java 17, but it can’t compile using Java 17.

I also think it makes sense to align JanusGraph release 1.0.0 soon after updating to TinkerPop 3.7.0. 
I would imagine the process be something like this:
1. Update to TinkerPop 3.7.0.
2. Add multi-query support to .label step (if this gets merged: https://github.com/apache/tinkerpop/pull/2097 ).
3. Make Java 17 builds (similarly as we do with Java 11 builds) (this may require some code changes as some of the functionality of Java 8 isn’t supported in Java 17 anymore).
